app/knights_configuration.py

```diff
@@ -0,0 +1,51 @@
+# Class for knights configuration
+class KnightsConfig:
+
+    def __init__(self, name: str, hp: int, protection: int, power: int):
+        self.name = name
+        self.hp = hp
+        self.protection = protection
+        self.power = power
+
+    def versus(self, enemy):
+        self.hp -= enemy.power - self.protection
+        enemy.hp -= self.power - enemy.protection
+        if self.hp <= 0:
+            self.hp = 0
+        if enemy.hp <= 0:
+            enemy.hp = 0
+
+
+# Function that configurate knights with KnightsConfig class
+def knights_configuration(knights):
+
+    for key_name, properties in knights.items():
+
+        # apply name
+        name = properties["name"]
+
+        # configurate health
+        hp = properties["hp"]
+
+        # apply armour
+        protection = 0
+        for armour in properties["armour"]:
+            protection += armour["protection"]
+
+        # apply weapon
+        power = properties["power"] + properties["weapon"]["power"]
+
+        # apply potion if exist
+        if properties["potion"] is not None:
+            if "power" in properties["potion"]["effect"]:
+                power += properties["potion"]["effect"]["power"]
+
+            if "protection" in properties["potion"]["effect"]:
+                protection += properties["potion"]["effect"]["protection"]
+
+            if "hp" in properties["potion"]["effect"]:
+                hp += properties["potion"]["effect"]["hp"]
+

```

Comment:
These code duplications can be also fixed, using `stats = ("protection", "power", "hp")` and looping on `stats`